packages/common/src/utils/Helpers/convertSigner.ts

```diff
@@ -51,7 +50,17 @@ export const convertSigner = async (signer: SupportedSigner, skipChainIdCalls: b
       }
       // convert viems walletClient to alchemy's SmartAccountSigner under the hood
       resolvedSmartAccountSigner = new WalletClientSigner(walletClient, "viem");
-      rpcUrl = walletClient?.transport?.url ?? null;
+      rpcUrl = walletClient?.transport?.url ?? undefined;
+    } else if (signer as PrivateKeyAccount) {
```

Comment:
I think this should be `instanceof` instead of `as`

Reply:
You're casting every other signer to a PrivateKeyAccount here, - not just accounts.

Reply:
instanceof works with classes, it does not work with types

Reply:
> You're casting every other signer to a PrivateKeyAccount here, - not just accounts.

Not sure what do you mean as this is the last "else if", it will cast it only if signer is indeed of PrivateKeyAccount type, otherwise it will break and throw an error.

Reply:
This was changed, please review